orcid-commons/src/main/java/no/sikt/nva/orcid/commons/model/business/OrcidCredentials.java

```diff
@@ -1,118 +1,43 @@
-    private static final String ID_TOKEN = "idToken";
-    private static final String TOKEN_ID = "tokenId";
-    private static final String MODIFIED = "modified";
-    private static final String CREATED = "created";
-
-    @JsonProperty(ORCID)
-    private URI orcid;
-
-    @JsonProperty(ACCESS_TOKEN)
-    private String accessToken;
-
-    @JsonProperty(TOKEN_TYPE)
-    private String tokenType;
-
-    @JsonProperty(EXPIRES_IN)
-    private int expiresIn;
-
-    @JsonProperty(TOKEN_VERSION)
-    private String tokenVersion;
-
-    @JsonProperty(PERSISTENT)
-    private boolean persistent;
-
-    @JsonProperty(ID_TOKEN)
-    private String idToken;
-
-    @JsonProperty(TOKEN_ID)
-    private int tokenId;
-
-    @JsonProperty(MODIFIED)
-    private Instant modified;
-
-    @JsonProperty(CREATED)
-    private Instant created;
-
-    @JacocoGenerated
-    public OrcidCredentials() {
-    }
-
-    public OrcidCredentials copy() {
-        return builder()
-                   .withOrcid(this.getOrcid())
-                   .withAccessToken(this.getAccessToken())
-                   .withTokenType(this.getTokenType())
-                   .withExpiresIn(this.getExpiresIn())
-                   .withTokenVersion(this.getTokenVersion())
-                   .withPersistent(this.isPersistent())
-                   .withIdToken(this.getIdToken())
-                   .withTokenId(this.getTokenId())
-                   .withModified(this.getModified())
-                   .withCreated(this.getCreated())
-                   .build();
-    }
-
-    @Override
-    @JacocoGenerated
-    public int hashCode() {
-        return Objects.hash(getOrcid(), getAccessToken(), getTokenType(), getExpiresIn(), getTokenVersion(),
-                            isPersistent(),
-                            getIdToken(), getTokenId(), getModified(), getCreated());
-    }
-
-    @Override
-    @JacocoGenerated
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (!(o instanceof OrcidCredentials)) {
-            return false;
-        }
-        OrcidCredentials that = (OrcidCredentials) o;
-        return hasSameCredentials(that)
-               && Objects.equals(getModified(), that.getModified())
-               && Objects.equals(getCreated(), that.getCreated());
-    }
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+@JsonSerialize
+public record OrcidCredentials(URI orcid,
+                               String accessToken,
+                               String tokenType,
+                               int expiresIn,
+                               String tokenVersion,
+                               boolean persistent,
+                               String idToken,
+                               int tokenId,
+                               Instant modified,
+                               Instant created) implements JsonSerializable {
```

Comment:
Since we're on the topic, we have 10 fields in this record, which leads me to believe that we're missing something in the design…e.g. we have a boolean "persistent" and a field "expiresIn"…are these related?

Reply:
The PR clearly explains the benefits of records, great work. We should have a quick run through of records + serialisation of these on Friday, please :) 

I appreciate the PR, btw. It's unfortunate that we have to rely on personal interest and Friday evenings to make these BIG improvements to the code, but this definitely brings the benefits of moving to Java 17 front-and-centre. Thank you :) make sure you submit the hours on your timesheet.

Reply:
Regarding the properties names: yes they are terrible. 

This is how they are served from Orcid (so yes there is a tokenId and an IdToken best namin ever!), and the documentation of what we are supposed to store doesn't match any of the the properties names we  received. So version 1: just store them as is.